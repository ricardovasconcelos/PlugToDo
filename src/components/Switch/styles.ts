import styled from 'styled-components';

export const SwithContainer = styled.div`
  display: flex;
  justify-content: space-between;

  p {
    font-weight: bold;
    color: ${(props) => props.theme['gray-800']};
  }
`;

export const SwitchWrapper = styled.div`
  position: relative;
`;

export const SwitchItem = styled.label`
  width: 2.3125rem;
  height: 1rem;

  position: absolute;
  top: 0;
  left: 0;

  border-radius: 15px;
  background-color: ${(props) => props.theme['gray-300']};
  cursor: pointer;

  &::after {
    content: '';
    display: block;

    width: 1.25rem;
    height: 1.25rem;

    border-radius: 50%;
    margin-top: -2px;
    background-color: ${(props) => props.theme['gray-100']};
    box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.1);

    transition: 0.2s;
  }
`;

export const SwitchCheckbox = styled.input`
  width: 2.625rem;
  height: 1.625rem;

  opacity: 0;
  z-index: 1;
  border-radius: 15px;

  &:checked + ${SwitchItem} {
    background: ${(props) => props.theme['gray-600']};
    &::after {
      content: '';
      display: block;
      width: 1.25rem;
      height: 1.25rem;

      border-radius: 50%;
      background: ${(props) => props.theme['blue-100']};

      margin-left: 21px;
      transition: 0.2s;
    }
  }
`;
