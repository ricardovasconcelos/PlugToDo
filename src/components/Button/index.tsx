import { ButtonHTMLAttributes, ReactNode } from 'react';

import { ButtonComponent } from './styles';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  width: number;
}

export function Button({ children, ...props }: ButtonProps) {
  return <ButtonComponent {...props}>{children}</ButtonComponent>;
}
