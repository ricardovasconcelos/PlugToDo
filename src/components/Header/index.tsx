import { HeaderContainer } from './styles';
import logo from '../../assets/logo.svg';
import { Link } from 'react-router-dom';

export function Header() {
  return (
    <HeaderContainer>
      <Link to='/'>
        <img src={logo} alt='Plug Logo' />
      </Link>
    </HeaderContainer>
  );
}
