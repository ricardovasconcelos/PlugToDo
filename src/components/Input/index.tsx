import { InputHTMLAttributes } from 'react';

import { InputComponent } from './styles';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {}

export function Input({ ...props }: InputProps) {
  return <InputComponent {...props} />;
}
