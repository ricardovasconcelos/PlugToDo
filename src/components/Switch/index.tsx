import { InputHTMLAttributes } from 'react';
import { SwithContainer, SwitchWrapper, SwitchCheckbox, SwitchItem } from './styles';

interface SwitchProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  checked?: boolean;
}

export function Switch({ label, checked, ...props }: SwitchProps) {
  return (
    <SwithContainer data-testid='switch-container'>
      <p>{label}</p>

      <SwitchWrapper>
        <SwitchCheckbox id='checkbox' type='checkbox' checked={checked} {...props} />
        <SwitchItem htmlFor='checkbox' />
      </SwitchWrapper>
    </SwithContainer>
  );
}
