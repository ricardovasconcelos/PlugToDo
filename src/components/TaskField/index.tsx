import { TaskFieldContainer, TaskDetailContainer } from './styles';
import { Task } from '../../db/db.types';

interface TaskFieldProps extends Omit<Task, 'id'> {
  onClickTask: () => void;
}

export function TaskField({ title, description, done, onClickTask }: TaskFieldProps) {
  return (
    <TaskFieldContainer done={done} onClick={onClickTask}>
      <TaskDetailContainer>
        <h3>{title}</h3>
        <p>{description}</p>
      </TaskDetailContainer>
    </TaskFieldContainer>
  );
}
