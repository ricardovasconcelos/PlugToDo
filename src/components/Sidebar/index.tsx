import { useRef, useEffect, useCallback, MouseEvent } from 'react';
import { useSpring, animated } from 'react-spring';
import { Button } from '../Button';
import { Input } from '../Input';

import {
  SidebarContainer,
  SidebarWrapper,
  SidebarContent,
  SidebarHeader,
  CloseSidebar,
  SidebarFooter,
} from './styles';

import close from '../../assets/close.svg';

interface SidebarProps {
  showSidebar: boolean;
  onShowSidebar: (value: boolean) => void;
}

export const Sidebar = ({ showSidebar, onShowSidebar }: SidebarProps) => {
  const sidebarRef = useRef<HTMLDivElement>(null);

  const animation = useSpring({
    config: {
      duration: 250,
    },
    opacity: showSidebar ? 1 : 0,
    transform: showSidebar ? `translateY(0%)` : `translateY(-100%)`,
  });

  const closeSidebar = (event: MouseEvent<HTMLDivElement>) => {
    if (sidebarRef.current === event.target) {
      onShowSidebar(false);
    }
  };

  const keyPress = useCallback(
    (event: KeyboardEvent) => {
      if (event.key === 'Escape' && showSidebar) {
        onShowSidebar(false);
      }
    },
    [onShowSidebar, showSidebar],
  );

  useEffect(() => {
    document.addEventListener('keydown', keyPress);
    return () => document.removeEventListener('keydown', keyPress);
  }, [keyPress]);

  return (
    <>
      {showSidebar ? (
        <SidebarContainer onClick={closeSidebar} ref={sidebarRef}>
          <animated.div style={animation}>
            <SidebarWrapper>
              <SidebarHeader>
                <CloseSidebar onClick={() => onShowSidebar(false)}>
                  <img src={close} alt='Botão para fechar modal de tarefas' />
                </CloseSidebar>
                <p>Nova Tarefa</p>
              </SidebarHeader>

              <SidebarContent>
                <Input placeholder='Título' />

                <Input placeholder='Descrição' />
              </SidebarContent>

              <SidebarFooter>
                <Button width={20}>CRIAR TAREFA</Button>
              </SidebarFooter>
            </SidebarWrapper>
          </animated.div>
        </SidebarContainer>
      ) : null}
    </>
  );
};
