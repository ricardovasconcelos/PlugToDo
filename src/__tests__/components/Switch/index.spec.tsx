import { screen, render, waitFor, fireEvent } from '@testing-library/react';

import { Switch } from '../../../components/Switch';

describe('Header', () => {
  it('should be able to render a Header', () => {
    const { container } = render(<Switch label='switch label' />);

    const SwitchContainer = screen.getAllByTestId('switch-container');
    waitFor(() =>
      expect(SwitchContainer).toHaveStyle({
        display: 'flex',
        justifyContent: 'space-between',
      }),
    );

    waitFor(() => expect(SwitchContainer).toBeInTheDocument());

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Changes its value when click', async () => {
    render(<Switch label='switch label' />);

    const switchButton = screen.getByRole('checkbox') as HTMLInputElement;
    fireEvent.click(switchButton);

    await waitFor(() => {
      expect(switchButton.checked).toEqual(true);
    });
  });
});
