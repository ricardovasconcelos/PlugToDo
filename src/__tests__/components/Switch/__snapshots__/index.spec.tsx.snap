// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able to render a Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
}

.c1 {
  position: relative;
}

.c4 {
  width: 2.3125rem;
  height: 1rem;
  position: absolute;
  top: 0;
  left: 0;
  border-radius: 15px;
  cursor: pointer;
}

.c4::after {
  content: '';
  display: block;
  width: 1.25rem;
  height: 1.25rem;
  border-radius: 50%;
  margin-top: -2px;
  box-shadow: 1px 3px 3px 1px rgba(0,0,0,0.1);
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c2 {
  width: 2.625rem;
  height: 1.625rem;
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
}

.c2:checked + .c3::after {
  content: '';
  display: block;
  width: 1.25rem;
  height: 1.25rem;
  border-radius: 50%;
  margin-left: 21px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<div
  class="c0"
  data-testid="switch-container"
>
  <p>
    switch label
  </p>
  <div
    class="c1"
  >
    <input
      class="c2"
      id="checkbox"
      type="checkbox"
    />
    <label
      class="c3 c4"
      for="checkbox"
    />
  </div>
</div>
`;
